### ISSUE TITLE
Migration to tofu1.7.0 from terraform1.8.2 shows updated in-place required for resources even though nothing has changed

### ISSUE THREAD

**Issue by mccoyer001** [opentofu/opentofu#1616]
### OpenTofu Version

```shell
terraform -v
Terraform v1.8.2
on windows_amd64
+ provider registry.terraform.io/hashicorp/azurerm v3.85.0
+ provider registry.terraform.io/hashicorp/random v3.6.1

tofu -v
OpenTofu v1.7.0
on windows_amd64
+ provider registry.opentofu.org/hashicorp/azurerm v3.85.0
+ provider registry.opentofu.org/hashicorp/random v3.6.1
```

### OpenTofu Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source = "hashicorp/azurerm"
      version = "=3.85.0"
    }
  }
}

# 2. Configure the AzureRM Provider
provider "azurerm" {
  skip_provider_registration = true
  features {}
}

data "azurerm_resource_group" "tofu-rg" {
  name     = "testrg"

}

data "azurerm_key_vault" "tofu-kv" {
  name                = "tofu-kv"
  resource_group_name = data.azurerm_resource_group.tofu-rg.name
}

resource "azurerm_key_vault_secret" "sample1" {
  name         = "SAMPLE1"
  value        = random_password.password1.result
  key_vault_id = data.azurerm_key_vault.tofu-kv.id
}

resource "random_password" "password1" {
  length           = 12
  special          = false
}
```

### Debug Output

https://gist.github.com/mccoyer001/2f16280ecff58252edd4f25b4d2d2b24

### Expected Behavior

"tofu plan" should not detect any changes required since the "terraform plan" shows no changes detected.

### Actual Behavior

tofu plan shows the need for updated in-place

### Steps to Reproduce

# initialize the terraform env and apply configuration and generate local terraform.tfstate file 
1. terraform init
2. terraform plan
3. terraform apply
# cleanup the terraform registry files 
4. del .\.terraform
5. del .\.terraform.lock.hcl
# initialize the tofu env and validate config has not changed
6. tofu init
7. tofu plan

### Additional Context

Have also seen this "updated in-place" false reporting for azure key vault secrets in my other setups.

**Comment by cam72cam** [comment]
Reproduced with just a small part of the example:
```
resource "random_password" "password1" {
  length           = 12
  special          = false
}
```

**Comment by Yantrio** [comment]
Small update:

I've been able to reproduce this locally, and I am currently investigating more.

During investigating this I have found that the `sensitive_attributes` are not being updated, that is correct. However that is not the cause of this issue. After fixing that issue, update in-place still happens.

I will put up another issue + PR to ensure that sensitive_attributes are propagated correctly from the schema soon and carry on investigating the cause of this issue for now.
